caas/kubernetes/provider/customresourcedefinitions.go

```diff
@@ -170,7 +237,11 @@ func (k *kubernetesClient) ensureCustomResources(
 			if err != nil {
 				return cleanUps, errors.Trace(err)
 			}
-			crSpec.SetLabels(k.getCRLabels(appName))
+			labels := k.getCRDLabelsGlobal(appName)
+			if isCRDNameSpaced(crd.Spec.Scope) {
+				labels = k.getCRDLabelsNamespaced(appName)
```

Comment:
Confusing that we use getCRDLabels for a CR

Reply:
@ycliuhw I think this comment may still need to be addressed with trying to isolate this logic to one place? Could be wrong.